addons/test_mail_full/tests/test_portal.py

```diff
@@ -38,6 +40,45 @@ def setUp(self):
 
         self.record_portal._portal_ensure_token()
 
+    def test_portal_message_fetch(self):
+        """Test retrieving chatter messages through the portal controller"""
+        self.authenticate(None, None)
+        message_fetch_url = '/mail/chatter_fetch'
+        payload = {
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'id': 0,
+            'params': {
+                'res_model': 'mail.test.portal',
+                'res_id': self.record_portal.id,
+                'token': self.record_portal.access_token,
+            },
+        }
+        post_headers = {'Content-Type': 'application/json'}
+
+        def get_chatter_message_count():
+            res = self.url_open(url=message_fetch_url, data=json.dumps(payload), headers=post_headers)
```

Comment:
By "inlined", I meant:

```py
res = self.url_open(
    url=message_fetch_url,
    data=json.dumps(payload),
    headers={'Content-Type': 'application/json'}
)
```

Sorry for the confusion ^^

Reply:
Oops, done. I also moved the `json.dumps` outside the method